apps/server/src/shared/domain/rules/task.rule.ts

```diff
@@ -19,52 +19,44 @@ export class TaskRule extends BasePermission<Task> {
 	}
 
 	public hasPermission(user: User, entity: Task, context: IPermissionContext): boolean {
-		const { action, requiredPermissions } = context;
-		const hasPermission = this.utils.hasAllPermissions(user, requiredPermissions);
-		const isCreator = this.utils.hasAccessToEntity(user, entity, ['creator']);
-
-		let hasTaskPermission = false;
-
-		if (action === Actions.read) {
-			hasTaskPermission = this.taskReadPermission(user, entity);
-		} else if (action === Actions.write) {
-			hasTaskPermission = this.taskWritePermission(user, entity);
+		let { action } = context;
+		const { requiredPermissions } = context;
+		const hasRequiredPermission = this.utils.hasAllPermissions(user, requiredPermissions);
+		if (!hasRequiredPermission) {
+			return false;
 		}
 
-		const result = hasPermission && (isCreator || hasTaskPermission);
-
-		return result;
-	}
-
-	private taskReadPermission(user: User, entity: Task): boolean {
-		const isDraft = entity.isDraft();
-		let hasParentReadPermission = false;
+		const isCreator = this.utils.hasAccessToEntity(user, entity, ['creator']);
+		const isAssigned = this.utils.hasAccessToEntity(user, entity, ['users']);
 
-		if (isDraft) {
-			hasParentReadPermission = this.parentPermission(user, entity, Actions.write);
-		} else {
-			hasParentReadPermission = this.parentPermission(user, entity, Actions.read);
+		if (entity.isDraft()) {
+			action = Actions.write;
 		}
 
-		const result = hasParentReadPermission;
-
-		return result;
-	}
+		const hasParentPermission = this.hasParentPermission(user, entity, action);
 
-	private taskWritePermission(user: User, entity: Task): boolean {
-		const hasParentWritePermission = this.parentPermission(user, entity, Actions.write);
+		// TODO why parent permission is with OR cond?
```

Comment:
or you asking yourself about what you implemented or to which permission is that question targeted? ^^

Reply:
it's exactly a TODO, because here is a piece of code which does not make sense, while out of scope in the current ticket to fix it

Reply:
Agree, maybe a follow up ticket creation would make sense then